src/main/java/strategy/SupplyOperation.java

```
@@ -0,0 +1,11 @@
+package strategy;
+
+import db.Storage;
+import model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.addQuantity(transaction.getFruit(), transaction.getQuantity());
```

The method calls into `Storage.addQuantity(...)` without catching/wrapping exceptions to add context. The requirement states: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and original cause." Wrap any exceptions from storage updates with a RuntimeException that includes transaction details (fruit, quantity, operation) and the original cause so failures are easy to diagnose.